### PR TITLE
feat: create webhooks

### PR DESCRIPTION
**Pull request by helenakallekleiv**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -8,6 +8,26 @@ resource "azurerm_container_registry" "this" {
...
 
+resource "azurerm_container_registry_webhook" "this" {
+  for_each = { for object in var.webhook : object.name => object }
+
+  depends_on = [
+    azurerm_container_registry.this
+  ]
+
+  name                = each.value.name
+  registry_name       = var.registry_name
```
Comment (hknutsen) [inline main.tf:19]: Get registry name from the container registry resource:

```suggestion
  name                = each.value.name
  registry_name       = azurerm_container_registry.this.name
```

This way, Terraform will create an automatic dependency between the webhook resources and the registry resource, removing the need of using the `depends_on` argument.

File: examples/webhook/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Webhook example
+
+Terraform configuration which creates an Azure Container Registry Webhook with the following features:
```
Comment (nemanja-drobac) [inline examples/webhook/README.md:3]:
```suggestion
Terraform configuration which creates an Azure Container Registry included Webhook resource with the following features:
```

File: variables.tf [hunk 3]
```diff
@@ -18,6 +18,18 @@ variable "sku" {
...
+  type = list(object({
+    name        = string
+    service_uri = string
+    status      = string
+    scope       = string
+    actions     = list(string)
+  }))
+  default = []
+}
+
```
Comment (nemanja-drobac) [inline variables.tf:32]:
```suggestion
variable "webhook" {
  description = "A list of objects required for this Webhook."
  type = list(object({
    name        = string
    service_uri = string
    actions     = list(string)
    status      = optional(string)
    scope       = optional(string)
  }))
  default = []
}

```

File: variables.tf [hunk 4]
```diff
@@ -18,6 +18,18 @@ variable "sku" {
...
+  description = "A list of objects required for this Webhook."
+  type = list(object({
+    name        = string
+    service_uri = string
+    status      = string
+    scope       = string
+    actions     = list(string)
+  }))
+  default = []
+}
```
Comment (hknutsen) [inline variables.tf:31]: Update variable name and description to better reflect that you can create multiple webhooks, and set type to map of objects and make some properties optional with default values:

```suggestion
variable "webhooks" {
  description = "A map of webhooks to create for this Container Registry."
  type = map(object({
    name        = string
    service_uri = string
    actions     = optional(list(string), ["push"])
    status      = optional(string, "enabled")
    scope       = optional(string, "")
  }))
  default = {}
```

By using maps, you could create two webhhooks `web` and `sql` using the following syntax:

```terraform
module "acr" {
  # source and arguments

  webhooks = {
    "web" = {
      # webhook properties
    }
    "sql" = {
      # webhook properties
    }
  }
}
```

Terraform would then create two webhook resources:

1. `module.acr.azurerm_container_registry_webhook.web`
2. `module.acr.azurerm_container_registry_webhook.sql`

This way, you have full control over the names of resources in your Terraform state file.

This is the same way we've done it for the other modules 🙂

File: examples/webhook/main.tf [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  webhook = [
+    {
+      name        = "mywebhook"
+      service_uri = "https://mywebhookreceiver.example/mytag"
+      status      = "enabled"
+      scope       = "mytag:*"
+      actions     = ["push"]
+    }
+  ]
```
Comment (hknutsen) [inline examples/webhook/main.tf:39]: Update according to my suggested changes in `main.tf`:

```suggestion
  webhooks = {
    "this" = {
      name        = "mywebhook"
      service_uri = "https://mywebhookreceiver.example/mytag"
      actions     = ["push"]
      status      = "enabled"
      scope       = "mytag:*"
    }
  }
```

**Review by hknutsen — CHANGES_REQUESTED** [review]
Looks good 🙂 Left a few comments to make it consistent with other modules.

**Comment by helenakallekleiv** [comment]
Go test passed on my local machine for all examples after committing suggestions.

**Review by hknutsen — APPROVED** [review]
Tests pass on my local machine.

Great work! 🙂

**Pull request merged**